cpp/src/structure/detail/structure_utils.cuh

```diff
@@ -83,6 +83,38 @@ struct rebase_offset_t {
   __device__ edge_t operator()(edge_t offset) const { return offset - base_offset; }
 };
 
+template <typename vertex_t, typename edge_t>
+rmm::device_uvector<vertex_t> expand_sparse_offsets(raft::device_span<edge_t const> offsets,
+                                                    vertex_t base_vertex_id,
+                                                    rmm::cuda_stream_view stream_view)
+{
+  edge_t num_edges{0};
+
+  if (offsets.size() > 0) {
+    raft::update_host(&num_edges, offsets.data() + offsets.size() - 1, 1, stream_view);
+  }
+
+  rmm::device_uvector<vertex_t> vertices(num_edges, stream_view);
+
+  if (num_edges > 0) {
+    thrust::fill(rmm::exec_policy(stream_view), vertices.begin(), vertices.end(), vertex_t{0});
+
+    raft::update_device(vertices.data(), &base_vertex_id, 1, stream_view);
+
+    thrust::for_each(
+      rmm::exec_policy(stream_view),
+      offsets.begin() + 1,
+      offsets.end(),
+      [d_vertices = vertices.data(), n_vertices = vertices.size()] __device__(auto offset) {
+        if (offset < n_vertices) { atomicAdd(&d_vertices[offset], vertex_t{1}); }
```

Comment:
Shouldn't this (offset < n_vertices) be always true? I guess this check is unnecessary.

Reply:
Oh, yes, you're right.